lib/matplotlib/widgets.py

```diff
@@ -1088,22 +1146,47 @@ def set_active(self, index):
         if self.drawon:
             self.ax.figure.canvas.draw()
 
-        if not self.eventson:
-            return
-        for cid, func in self.observers.items():
-            func(self.labels[index].get_text())
+        self._exec_callbacks(self.labels[index].get_text())
+
+    def clear(self):
+        """Deactivate any previously activated button."""
+        if self.index_selected is not None:
+            ax_facecolor = self.ax.get_facecolor()
+            self.circles[self.index_selected].set_facecolor(ax_facecolor)
+        self.value_selected = None
+        self.index_selected = None
+
+        # calling it with no label, as all buttons are being cleared
+        self._exec_callbacks(None)
 
     def on_clicked(self, func):
         """
-        Connect the callback function *func* to button click events.
+        Set callback function.
 
-        Returns a connection id, which can be used to disconnect the callback.
+        Parameters
+        ----------
+        func : callable
+            When the button is clicked, call *func* with button label
```

Comment:
See https://matplotlib.org/devdocs/api/_as_gen/matplotlib.artist.Artist.add_callback.html?highlight=signature for a good way to describe a callable.

Reply:
Good pointer, used it to improve docs.